test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+    cellularStatus = Cellular_ATRemoveAllDoubleQuote( pString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATGetNextTok to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATGetNextTok_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString;
+    char * pTokOutput;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetNextTok( NULL, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetNextTok( &pString, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetNextTok( NULL, &pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATGetNextTok to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATGetNextTok_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_TOKEN_STRING_INPUT ) + 1 ) );
+    char * pStringSource = pString;
+    char * pToken;
+    strcpy( pString, CELLULAR_SAMPLE_TOKEN_STRING_INPUT );
+
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATGetNextTok( & pString,
+                                            & pToken );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    /* Test the result to check its functionality. */
+    TEST_ASSERT_EQUAL_STRING( CELLULAR_SAMPLE_TOKEN_OUTPUT, pToken );
+
+    free( pStringSource );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATGetSpecificNextTok to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATGetSpecificNextTok_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString;
+    const char * pDelimiter;
+    char * pTokOutput;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( NULL, pDelimiter, &pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, NULL, &pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATGetSpecificNextTok to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATGetSpecificNextTok_Happy_Path( void )

```

Comment:
This can be broken in two tests.

Reply:
sure, will divide this one into `test_Cellular_ATGetSpecificNextTok_WITH_DELIMTER_FIRST` and `test_Cellular_ATGetSpecificNextTok_WITHOUT_DELIMITER`.